packages/task/src/task.ts

```diff
@@ -423,18 +423,26 @@ export class Task<
     return this._error;
   }
 
-  render<T extends StatusRenderer<R>>(renderer: T) {
+  render<S extends StatusRenderer<R, T>>(renderer: S) {
     switch (this.status) {
       case TaskStatus.INITIAL:
-        return renderer.initial?.() as MaybeReturnType<T['initial']>;
+        return renderer.initial?.(this._previousArgs) as MaybeReturnType<
```

Comment:
`_previousArgs` doesn't sound quite right anymore if we're using them for the current render. Maybe it should be named `_latestArgs`?